src/plugins/intel_gpu/src/graph/network.cpp

```diff
@@ -913,6 +926,9 @@ void network::allocate_primitive_instance(program_node const& node) {
         if (node.is_type<data>())
             _data_outputs.push_back(inst);
     }
+    if (node.is_type<read_value>()) {
+        _read_values.push_back(inst);
```

Comment:
I think this is redundant, Iterating primitives are O(N) so it would be enough to iterate primitives and do the above procedure if it is readvalue.

Reply:
applied as comment